blog-post.md

```diff
@@ -0,0 +1,292 @@
+provides complete type safety without explicit annotations.
+
+```
+let constant_value = 42; // Inferred as word
+let heterogeneous_tuple = (address(0), true, 42); // Inferred as (address, bool, word)
+constant_value = heterogeneous_tuple ; // this is a type error!
+```
+
+
+
+### Core Solidity and SAIL
+
+The combination of all these features do form a complete programming language, 
+which we call SAIL (Solidity Abstract Intermediate Language). Core Solidity is 
+built on top of SAIL by providing more specialized syntactic constructs that 
+replicate familiar features from Classic Solidity, alongside a mechanism for 
+"desugaring" them into SAIL equivalent code. But what Core Solidity features 
+will be translated into SAIL?
+
+- Data Locations represented as types: Data locations (e.g., storage, memory) 
+are now a part of a type. This enables the creation of composite types with 
+mixed locations, such as a struct that holds references to storage 
+arrays and memory data, as presented in the next code piece:
+
+```
+struct Broker {
+  total : memory(word) 
+, accounts : storage(array(address, word))
+}
+```
+
+- Static/dynamic array - literals: Classic Solidity limits array literals to static 
+arrays. Core Solidity type system allows literals to adopt the correct type (static 
+or dynamic) from their usage context from the start.
+
+The current Core Solidity prototype support all previous features. In the near 
+future, we plan to develop: 
+
+- Compile time evaluation: Inspired by Zig, we plan to include compile-time code 
+evaluation in Core Solidity. Details on how such feature will work are being 
+discussed by Argot Collective Programming Languages research team.
+
+- Redesigned Module System: Currently, the Core Solidity prototype does not have 
+a proper module system. In our tests, we implemented a import directive which 
+behaves like C's `#include`. The design of a proper module system for Core Solidity 
+are being discussed by Argot Collective Programming Languages research team.
+
+- Macros: One of the main design goals of Core Solidity is to reduce the burden 
+of extending and maintaining the Solidity Compiler. Currently, the Core Solidity 
+prototype has a internal implementation of all desugaring steps into SAIL into the 
+compiler itself, which conflicts with our design goal of having the most of language 
+features being part of the standard library itself. One way to achieve such goal 
+is by having a powerful **macro** system, like Lean 4 or modern Lisp dialects such 
+as Racket. Having macros, most of the desugaring steps would be implemented by 
+macros, allowing such steps to be part of libraries, giving users the power 
+to replace standard library implementations to code that better suit their needs.
+The exact details of how such macro system will be discussed in the near future 
+by Argot Collective Programming Languages research team.
```

Comment:
Wasn't the macro system something to be done only post 1.0? Or will we actually be designing it in the near future?